### ISSUE TITLE
MacOS: Crashing randomly when working with a video

### ISSUE THREAD

**Issue by lorilay7** [Aegisub/Aegisub#65]
Aegisub r8898 on MacOS X 10.13.4 crashes randomly when translating with a video, after moving from one line to the other or moving the slider along the spectrum display. With larger video files it's happening more often than with the smaller ones.

**Comment by CoffeeFlux** [comment]
Does this happen with any video file, or just a particular one? Can you give any info from the crashes?

**Comment by lorilay7** [comment]
Any video file. And the program just closes no error message except for Apple standard one.
Tell me what info do you need, and I'll provide it.

**Comment by lorilay7** [comment]
This is the full log from when I'm moving the slider across the audio spectrum.
[Crash log 1.txt](https://github.com/Aegisub/Aegisub/files/2035095/Crash.log.1.txt)

**Comment by lorilay7** [comment]
And this is from moving between lines
[Crash log 2.txt](https://github.com/Aegisub/Aegisub/files/2042560/Crash.log.2.txt)

**Comment by CoffeeFlux** [comment]
I'm able to reproduce this in my vm at least. I'll look into the cause later.

**Comment by lorilay7** [comment]
The bug still present in r8942.

**Comment by nbo21** [comment]
I think this issue really is audio related. 

With the audio closed I can move quickly from line to line and there are no crashes from start to end. But as soon as I load the audio and try to do the same it crashes after a very short time. 

I have tested this with a handful of video files that all contain AAC audio.

**Comment by DanLabute** [comment]
I dug into this a but more in 3.2.2 and I can get this to occur reliably using videos using H.264 encoded with FFMPEG, regardless of video length, and regardless of audio codec - the issue even happens with the video loaded but the audio closed.

If I re-encode the video using avidemux's VideoToolbox H.264 codec and copy the audio stream, I can use this re-encoded video + audio in Aegisub without any crashes occurring.

Seems fairly definitive that this crash is FFMPEG H.264 related - are there are log files I can provide that would assist with tracking this down?

**Comment by NoBodyButMe** [comment]
Still the same problem. I'm moving to another solution. This is... not funny. Starting over and over again. Always losing my work.

**Comment by XStylus** [comment]
I don't think this issue is limited to H.264s encoded with FFMPEG. I've encountered this issue with non-H264 videos, and also H.264 videos encoded using Adobe Media Encoder, as well as Quicktime H264 MOVs.

I'm encountering this issue on the plorkyeran build of Aegisub (r8942) on all versions of MacOS ranging from 10.12.6 on up to 10.14.6.

**Comment by wangqr** [comment]
There is a known bug in cache invalidation which will cause accessing invalid memory regions during freeing cache blocks. See https://github.com/wangqr/Aegisub/commit/9e6b7e94c04e5e76fc5d65829ab7f7f992c0b3df . The patch is already merged after #142 , but not in r8942. Has anyone tried the git master?

Alternatively you can try increase the cache size in `Preference -> Advanced -> Audio -> Spectrum -> Cache memory max`, and see if this delays or lowers the frequency of the crash.

**Comment by joenca** [comment]
For me this bug was triggered when using a USB audio device for sound output. Switching output device to Internal Speakers fixed the crashing.

**Comment by Sesamme** [comment]
I just had regular crashes at one specific location of the video. The file was H.264, ACC encoded. I tried to increase the cash size, but it didn't help. So I have recompressed it with Quicktime using H.264 but much smaller dimentions than original, and Apple Lossless Audio instead of ACC, and it stopped crashing. Don't know what was the cause.

**Comment by majinzero** [comment]
Hello,
I have the same problem, when I want to change a certain video to a certain timecode the program crashes.
[Crash.txt](https://github.com/Aegisub/Aegisub/files/6733328/Crash.txt)
Maybe you can fix this problem.

**Comment by Sesamme** [comment]
@majinzero

Try recompressing your video with different settings (make it smaller for example)

**Comment by majinzero** [comment]
i download a new aegisub version now it works.
